### ISSUE TITLE
Windows 9x/MSVC - possible inconsistent matches

### ISSUE THREAD

**Issue by ItzSwirlz** [decompme/decomp.me#854]
TL;DR - The stack/registers used in one function can modify the stack/registers in later functions. This may be a LEGO Island specific issue but could also be a limitation with MSVC.
---
LEGO Island at least, is a DLL - I'm not sure if it happens in other games or games that are EXEs and not DLLs, but often changing things in one function can result in changing the registers or values of another.

For example, although in this PR about MxBitmap [vtable functions](https://github.com/isledecomp/isle/pull/89) although I didn't touch `MxDirectDraw` at all, let alone `MxDirectDraw::ErrorToString`, the matching of it changed.
<img width="791" alt="image" src="https://github.com/decompme/decomp.me/assets/24401303/003ba088-f728-45eb-bea2-768e764be553">

Another example is looking at ISLE.exe - LEGO Island was deployed to have a simple frontend for keyboard inputs, game initialization, etc. in a small executable, and the rest of the game is LEGO1.DLL and other scripts. Although ISLE is completely done, and technically should be 100%, the matching of it actually fluctuates between 95-100%.

The point I'm getting at here is that MSVC-resulting assembly in other functions are altered by each other - that is, a change in the offsets/additions like `-je 0x14f` vs `+je 0x13d` can impact other functions.

I worked on trying to match MxPalette::operator== (https://decomp.me/scratch/8ehhi), however the match (and even the diff when originally imported) was different than what was exported. My best explanation is that MxPalette::SetEntries, which in the LEGO1.DLL has its function offset placed before MxPalette::operator==, is incorrect (which it is, it has register swaps and some incorrect MOVs and ADDs), so SetEntries would need to be matched first before seeing if `operator==` is an issue.

Generally you'll see this by simple register swaps, or the location of some instructions being moved around. Sometimes it's simple things like `mov bl, byte ptr [ecx + 1]` vs, idk, `mov bl, [ecx + 1]` or something small like that.

While it is possible that LEGO Island's compiler flags are wrong, the fact they don't match with the rest of the code is something that sort of is annoying, because then for best results, matching MxPalette::operator== would result in just importing and matching the assembly for the ENTIRE class.

However, if a function obviously isn't changing in its percentage/assembly, and is always bad (like MxBitmap::ImportColorsToPalette) then that's definitely something decomp.me can help with

**Comment by ItzSwirlz** [comment]
The build log for this PR also is a good example of showing how altering a few things results in alterations of other functions/classes: https://github.com/isledecomp/isle/actions/runs/6197219147/job/16825407427

**Comment by mkst** [comment]
I'm not sure what we as maintainers of decomp.me are meant to do with this information?

**Comment by ethteck** [comment]
Echoing what I said on discord:

Decomp "gotchas" are important to be aware of, but unless you had something tangible in mind the site should be doing about this, an issue isn't really the appropriate place for it. However, it's great you all are making important discoveries about the compiler and how things can affect matching.
